### PR TITLE
[Event Hubs] Name updates for event processor types

### PR DESCRIPTION
**Pull request by srnagar**
This is the first set of name updates based on the API review feedback. This PR has name changes for
- EventProcessor -> EventProcessorClient
- EventProcessorStore -> CheckpointStore
- buildEventProcessor() -> buildEventProcessorClient()
- EventProcessorBuilder -> EventProcessorClientBuilder
- consumerGroupName -> consumerGroup

### CODE REVIEW

File: sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md [hunk 1]
```diff
@@ -88,16 +88,16 @@ provide, allowing your application to focus on the business logic needed to prov
...
 In our example, we will focus on building the [`EventProcessor`][source_eventprocessor], use the 
-[`BlobEventProcessorStore`][source_blobeventprocessorstore], and a simple callback function to process the events 
+[`BlobCheckpointStore`][source_blobcheckpointstore], and a simple callback function to process the events 
 received from the Event Hubs, writes to console and updates the checkpoint in Blob storage after each event.
 
 ```java
 class Program {
     public static void main(String[] args) {
-        EventProcessor eventProcessor = new EventProcessorBuilder()
+        EventProcessor eventProcessorClient = new EventProcessorBuilder()
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md]: Looks like EventProcessor still here

File: sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md [hunk 2]
```diff
@@ -144,14 +144,14 @@ Guidelines](./CONTRIBUTING.md) for more information.
...
 [performance_tuning]: https://github.com/Azure/azure-sdk-for-java/wiki/Performance-Tuning
-[sample_container_client]: ./src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobEventProcessorStoreSample.java
+[sample_container_client]: ./src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStoreSample.java
 [sample_event_hubs]: ./src/samples/java/com/azure/messaging/eventhubs
-[sample_event_processor]: ./src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobEventProcessorStoreSample.java
+[sample_event_processor]: ./src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobCheckpointStoreSample.java
 [sample_examples]: ./src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob
 [sas_token]: https://docs.microsoft.com/azure/storage/common/storage-dotnet-shared-access-signature-part-1
 [source_code]: ./
 [source_eventprocessor]: ./src/main/java/com/azure/messaging/eventhubs/EventProcessor.java
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/README.md]: The link to EventProcessor.java will be broken.

File: sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStore.java [hunk 3]
```diff
@@ -29,14 +29,14 @@
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 /**
- * Implementation of {@link EventProcessorStore} that uses
+ * Implementation of {@link CheckpointStore} that uses
  * <a href="https://docs.microsoft.com/en-us/azure/storage/common/storage-introduction#blob-storage">Storage Blobs</a>
- * for persisting partition ownership and checkpoint information. {@link EventProcessor EventProcessors} can use this
+ * for persisting partition ownership and checkpoint information. {@link EventProcessorClient EventProcessors} can use this
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStore.java]: Should this be EventProcessorClients vs EventProcessors?

File: sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobCheckpointStoreSample.java [hunk 4]
```diff
@@ -52,17 +52,17 @@ public static void main(String[] args) throws Exception {
...
             .processEvent(PARTITION_PROCESSOR)
             .processError(ERROR_HANDLER)
-            .eventProcessorStore(new BlobEventProcessorStore(blobContainerAsyncClient));
+            .checkpointStore(new BlobCheckpointStore(blobContainerAsyncClient));
 
-        EventProcessor eventProcessor = eventProcessorBuilder.buildEventProcessor();
-        eventProcessor.start();
+        EventProcessorClient eventProcessorClient = eventProcessorClientBuilder.buildEventProcessorClient();
+        eventProcessorClient.start();
         TimeUnit.MINUTES.sleep(5);
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/samples/java/com/azure/messaging/eventhubs/checkpointstore/blob/EventProcessorBlobCheckpointStoreSample.java]: nit: Add a comment that we wanted to simulate the customer's program doing other work until it is ready to stop.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java [hunk 5]
```diff
@@ -24,21 +25,21 @@
...
  * Processor may own zero or more partitions to balance the workload among all the instances.
  *
- * <p>To create an instance of EventProcessor, use the fluent {@link EventProcessorBuilder}.</p>
+ * <p>To create an instance of EventProcessor, use the fluent {@link EventProcessorClientBuilder}.</p>
  *
- * @see EventProcessorBuilder
+ * @see EventProcessorClientBuilder
  */
-public class EventProcessor {
+public class EventProcessorClient {
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java]: Add annotation for ServiceClient and ServiceClientBuilder in the other one.

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java [hunk 6]
```diff
@@ -50,27 +51,27 @@
      * @param consumerGroup The consumer group name used in this event processor to consumer events.
      * @param partitionProcessorFactory The factory to create new partition processor(s).
      * @param initialEventPosition Initial event position to start consuming events.
-     * @param eventProcessorStore The partition manager used for reading and updating partition ownership and checkpoint
+     * @param checkpointStore The partition manager used for reading and updating partition ownership and checkpoint
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventProcessorClient.java]:
```suggestion
     * @param checkpointStore The store used for reading and updating partition ownership and checkpoints.
```

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java [hunk 7]
```diff
@@ -26,10 +26,10 @@
 
 /**
  * This class is responsible for balancing the load of processing events from all partitions of an Event Hub by
- * distributing the number of partitions uniformly among all the  active {@link EventProcessor EventProcessors}.
+ * distributing the number of partitions uniformly among all the  active {@link EventProcessorClient EventProcessors}.
  * <p>
- * This load balancer will retrieve partition ownership details from the {@link EventProcessorStore} to find the number
- * of active {@link EventProcessor EventProcessors}. It uses the last modified time to decide if an EventProcessor is
+ * This load balancer will retrieve partition ownership details from the {@link CheckpointStore} to find the number
+ * of active {@link EventProcessorClient EventProcessors}. It uses the last modified time to decide if an EventProcessor is
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionBasedLoadBalancer.java]: EventProcessorClients?

File: sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/Checkpoint.java [hunk 8]
```diff
@@ -14,7 +14,7 @@
 public class Checkpoint {
```
Comment (conniey) [inline sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/Checkpoint.java]: iirc, Ramya wanted to add fullyQualifiedNamespace to this

Comment (srnagar) [reply sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/models/Checkpoint.java]: Added. I thought I'll add this in a later PR but included it in this PR itself now.

**Pull request merged**